packages/webapp/src/containers/Filter/Transactions/index.jsx

```diff
@@ -0,0 +1,86 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { EXPENSE_TYPE, REVENUE_TYPE } from '../constants';
+import { expenseTypeSelector } from '../../Finances/selectors';
+import { revenueTypesSelector } from '../../revenueTypeSlice';
+import FilterGroup from '../../../components/Filter/FilterGroup';
+import { Semibold } from '../../../components/Typography';
+import styles from './styles.module.scss';
+import { transactionTypeEnum } from '../../Finances/useTransactions';
+
+const TransactionFilterContent = ({ transactionsFilter, filterRef, onChange }) => {
+  const { t } = useTranslation(['translation', 'filter']);
+  const expenseTypes = useSelector(expenseTypeSelector);
+  const revenueTypes = useSelector(revenueTypesSelector);
+
+  const filters = [
+    {
+      subject: t('FINANCES.FILTER.EXPENSE_TYPE'),
+      filterKey: EXPENSE_TYPE,
+      options: [
+        ...expenseTypes.map((type) => ({
+          value: type.expense_type_id,
+          default: transactionsFilter[EXPENSE_TYPE]?.[type.expense_type_id]?.active ?? true,
```

Comment:
In reading the code (!) this was the hardest for me to comprehend because it didn't read logically that a default value would be conditional on the current filter state.

I guess it comes from what I would call `componentInitialState` being named `defaultFilterState` (I know it's not your naming convention!) Only if it was hard for you to catch the first time as well, it might be worth a comment here what `default` indicates.

Reply:
Yeah, this was confusing for me at first too!